tools/c7n_mailer/c7n_mailer/cli.py

```diff
@@ -47,39 +55,82 @@
 }
 
 
-def setup_parser():
-    parser = argparse.ArgumentParser()
-    parser.add_argument('-c', '--config', required=True)
-    return parser
-
-
-def session_factory(config):
+def session_factory(mailer_config):
     return boto3.Session(
-        region_name=config['region'],
-        profile_name=config.get('profile'))
-
+        region_name=mailer_config['region'],
+        profile_name=mailer_config.get('profile', None))
 
-def main():
-    parser = setup_parser()
-    options = parser.parse_args()
 
-    import logging
+def get_logger(debug=False):
     log_format = '%(asctime)s - %(name)s - %(levelname)s - %(message)s'
-    logging.basicConfig(level=logging.DEBUG, format=log_format)
+    logging.basicConfig(level=logging.INFO, format=log_format)
     logging.getLogger('botocore').setLevel(logging.WARNING)
+    if debug:
+        logging.getLogger('botocore').setLevel(logging.DEBUG)
+        debug_logger = logging.getLogger('custodian-mailer')
+        debug_logger.setLevel(logging.DEBUG)
+        return debug_logger
+    else:
+        return logging.getLogger('custodian-mailer')
 
-    with open(options.config) as fh:
-        config = yaml.load(fh.read(), Loader=yaml.SafeLoader)
 
+def get_and_validate_mailer_config(args):
+    with open(args.config) as fh:
+        config = yaml.load(fh.read(), Loader=yaml.SafeLoader)
     jsonschema.validate(config, CONFIG_SCHEMA)
     utils.setup_defaults(config)
+    return config
+
+
+def get_c7n_mailer_parser():
+    parser = argparse.ArgumentParser()
+    parser.add_argument('--config', required=True, help='mailer.yml config file')
+    debug_help_msg = 'sets c7n_mailer logger to debug, for maximum output (the default is INFO)'
+    parser.add_argument('--debug', action='store_true', help=debug_help_msg)
+    max_num_processes_help_msg = 'will run the mailer in parallel, integer of max processes allowed'
+    parser.add_argument('--max-num-processes', help=max_num_processes_help_msg)
+    group = parser.add_mutually_exclusive_group(required=True)
+    update_lambda_help_msg = 'packages your c7n_mailer, uploads the zip to aws lambda as a function'
+    group.add_argument('--update-lambda', action='store_true', help=update_lambda_help_msg)
+    run_help_msg = 'run c7n-mailer locally, process sqs messages and send emails or sns messages'
+    group.add_argument('--run', action='store_true', help=run_help_msg)
+    return parser
 
+
+def run_mailer_in_parallel(mailer_config, aws_session, logger, max_num_processes):
```

Comment:
@mwkaufman so the mailer previously could only do one thing, that is update lambda. Now it does multiple things, update, run, debug, multiprocessing. Just specifying the config option (with no other args), and causing that to update lambda by default seems like confusing behavior (especially for new users). It seems easy enough to just specify --update-lambda, or --run. mutative actions are potentially dangerous, so explicitly opting in to them seems reasonable. I'll leave it up to @kapilt to decide on this though.